File: love/sounds.lua
-- Blackjack
-- Copyright 2015 Sam Saint-Pettersen
-- Released under the MIT/X11 License.
--
-- Powered by the LÖVE Game Engine

--- SoundEffects class for Blackjack.
-- @copyright 2015 Sam Saint-Pettersen

s = 'sfx/'
soundOn = true

--- Toggle sound effects on/off.
-- @return [boolean] Are sound effects on?
function SoundEffects_toggle()
	if soundOn then
		soundOn = false
	else
		soundOn = true
	end
	return soundOn
end

--- Play sound effect.
-- @param [string] effect Name of sound effect to play.
function SoundEffects_play(effect)
	local e = love.audio.newSource(s .. effect .. '.ogg', 'static')
	love.audio.play(e)
end

File: html5/enchant.js_ts/sounds.ts
/// <reference path="enchant.d.ts" />

/* Sound effects for generated game. */
var s: string = 'sounds/';

class SoundEffects {
	private static soundOn = true;

	/**
	 * Toggle sound effects on/off.
	 * @returns {boolean} Are sound effects on?
	*/
	public static toggle() {
		if(SoundEffects.soundOn)
			SoundEffects.soundOn = false;
		else
			SoundEffects.soundOn = true;

		return SoundEffects.soundOn;
	}

	/**
	 * Play sound effect.
	 * @param {Core} game - Enchant.js game object.
	 * @param {string} effect - Name of sound effect to play.
	*/
	public static play(game: Core, effect: string) {
		if(SoundEffects.soundOn) {
			if (game.sfx)
				game.assets[s + effect + '.ogg'].play();
			else
				new Audio(s + effect + '.ogg').play();
		}
	}
}
